header/sync/sync.go

```diff
@@ -45,14 +48,18 @@ type Syncer struct {
 	pending ranges
 	// cancel cancels syncLoop's context
 	cancel context.CancelFunc
+
+	// isRunning shows service state
+	isRunning uint64
```

Comment:
For the Syncer this isn't actually needed if the goal is only to make Stop():
* Thread-safe
* Safe to be called multiple times. E.g. FP happened and then FX calls Stop

Reply:
Instead, we can remove setting `s.cancel` to nil in Stop + removing the check for nil `s.cancel` in Start.
It is also _safe_ to ignore the error coming from Subscriber.RemoveValidator or Subsciber.Stop(as suggested in another comment) in this case.

Reply:
Yes, good idea. Thanks.

Reply:
I removed flag as you proposed. So, now after receiving the BEFP syncer will call `Stop()` in P2PSubscriber